recipe/meta.yaml

```diff
@@ -39,14 +39,17 @@ requirements:
     - eigen
     - libboost-devel
     - qhull
+    # The constraint is on vtk-base instead of vtk otherwise conda-smithy does not pin vtk in .ci_support
+    # if it sees there is a different pin in the meta.yaml
+    - vtk-base * qt*
     - vtk
```

Comment:
FYI, for adding additional constraints, you can also use
```yaml
    - vtk * qt*
    - vtk
```
. The important part is that you provide one plain `vtk` dependency for which `conda-build` will do its variant handling (because, as you noticed, it won't consider variants/pinnings if version/build strings are constrained).
But it's also fine to have multiple dependencies on the same package, which is why you can add constraints in additional dependency entries.

Reply:
Ah cool, that is great and more clear then my solution, thanks!